convert2rhel/checks.py

```diff
@@ -57,6 +60,71 @@ def perform_pre_ponr_checks():
     ensure_compatibility_of_kmods()
 
 
+def check_convert2rhel_latest():
+    """Make sure that we a running the latest version of convert2rhel"""
+
+    # get lastest avalible version
+
+    repo_content = (
+        "[convert2rhel]\n"
+        "name=Convert2RHEL Repository\n"
+        "baseurl=https://ftp.redhat.com/redhat/convert2rhel/$releasever/os/\n"
+        "gpgcheck=0\n"
+        "enabled=1\n"
+    )
+
+    repo_dir = os.path.join(utils.TMP_DIR, "convert2rhel_repo")
+    # repo_dir = os.path.join("/tmp/","convert2rhel_repo")
+    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
+    store_content_to_file(filename=repo_path, content=repo_content)
+
+    # repoquery --setopt=reposdir=$(pwd)/test_repo --releasever=8 convert2rhel
+    raw_output_convert2rhel_versions, _ = run_subprocess(
+        ("repoquery --releasever={releasever} --setopt=reposdir={repodir} convert2rhel").format(
+            releasever=system_info.releasever,
+            # releasever="8",
+            repodir=repo_dir,
+        ),
+        print_output=False,
+    )
+
+    #   - if cannot access url: print warning and continue
+    PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
```

Comment:
@abadger or @SpyTec I may be wrong, but this regex should be compiled first and preferably outside the function isn't it?

Reply:
It could be, but not strictly necessary. It's a performance enhancement that makes more sense when you have to iterate through something and check with a RegEx multiple times, or expect the function to be called multiple times. I'm indifferent to whichever solution is done in this case

Reply:
Gotcha! Will take a note of that for future reviews. Thanks!